tests/spark/test_spark_connect_model_export.py

```diff
@@ -0,0 +1,182 @@
+
+
+def _get_spark_connect_session():
+    return (
+        SparkSession.builder.remote("local[2]")
+            .config("spark.connect.copyFromLocalToFs.allowDestLocal", "true")
+            .getOrCreate()
+    )
+
+
+@pytest.fixture
+def model_path(tmp_path):
+    return os.path.join(tmp_path, "model")
+
+
+def score_model_as_udf(model_uri, pandas_df, result_type):
+    spark = SparkSession.getActiveSession()
+    spark_df = spark.createDataFrame(pandas_df).coalesce(1)
+    pyfunc_udf = spark_udf(
+        spark=spark, model_uri=model_uri, result_type=result_type, env_manager="local"
+    )
+    new_df = spark_df.withColumn("prediction", pyfunc_udf(spark_f.struct(spark_f.col("features"))))
+    return new_df.toPandas()["prediction"]
+
+
+@pytest.fixture(scope="module")
+def spark():
+    spark = _get_spark_connect_session()
+    yield spark
+    spark.stop()
+
+
+@pytest.fixture(scope="module")
+def iris_df(spark):
+    iris = datasets.load_iris()
+    X = iris.data
+    y = iris.target
+
+    spark_df = spark.createDataFrame([
+        (features, label)
+        for features, label in zip(X, y)
+    ], schema="features: array<double>, label: long")
+
+    pandas_df = spark_df.toPandas()
+    return pandas_df, spark_df
+
+
+@pytest.fixture(scope="module")
+def spark_model_iris(iris_df):
+    iris_pandas_df, iris_spark_df = iris_df
+    scaler = StandardScaler(inputCol="features", outputCol="scaled_features")
+    lr = LORV2(maxIter=50, numTrainWorkers=2, learningRate=0.001)
+    pipeline = Pipeline(stages=[scaler, lr])
+    # Fit the model
+    model = pipeline.fit(iris_spark_df)
+    preds_pandas_df = model.transform(iris_pandas_df.copy(deep=False))
+    return SparkModelWithData(
+        model=model,
+        spark_df=iris_spark_df,
+        pandas_df=iris_pandas_df,
+        predictions=preds_pandas_df,
+    )
+
+
+@pytest.fixture
+def model_path(tmp_path):
+    return os.path.join(tmp_path, "model")
+
+
+def test_model_export(spark_model_iris, model_path):
+    mlflow.spark.save_model(spark_model_iris.model, path=model_path)
+    # 1. score and compare reloaded sparkml model
+    reloaded_model = mlflow.spark.load_model(model_uri=model_path)
+    preds_df = reloaded_model.transform(spark_model_iris.pandas_df.copy(deep=False))
+    pd.testing.assert_frame_equal(spark_model_iris.predictions, preds_df, check_dtype=False)
+
+    m = pyfunc.load_model(model_path)
+    # 2. score and compare reloaded pyfunc
+    preds2 = m.predict(spark_model_iris.pandas_df.copy(deep=False))
+    pd.testing.assert_series_equal(spark_model_iris.predictions["prediction"], preds2, check_dtype=False)
+
+    # 3. score and compare reloaded pyfunc Spark udf
+    preds3 = score_model_as_udf(model_uri=model_path, pandas_df=spark_model_iris.pandas_df, result_type=LongType())
+    pd.testing.assert_series_equal(spark_model_iris.predictions["prediction"], preds3, check_dtype=False)
+
+
+@pytest.mark.parametrize("should_start_run", [False, True])
```

Comment:
We don't need this. The model serializaiton/deserializaiton logic is unrelated to how we start/stop MLflow runs.